src/libraries/System.Text.RegularExpressions/gen/RegexGenerator.Emitter.cs

```diff
@@ -4666,11 +4666,25 @@ private static string MatchCharacterClass(RegexOptions options, string chExpr, s
 
             // We know that the whole class wasn't ASCII, and we don't know anything about the non-ASCII
             // characters other than that some might be included, for example if the character class
-            // were [\w\d], so if ch >= 128, we need to fall back to calling CharInClass, otherwise use
-            // the lookup table.
-            return negate ?
-                $"((ch = {chExpr}) < 128 ? ({Literal(bitVectorString)}[ch >> 4] & (1 << (ch & 0xF))) == 0 : !RegexRunner.CharInClass((char)ch, {Literal(charClass)}))" :
-                $"((ch = {chExpr}) < 128 ? ({Literal(bitVectorString)}[ch >> 4] & (1 << (ch & 0xF))) != 0 : RegexRunner.CharInClass((char)ch, {Literal(charClass)}))";
+            // were [\w\d], so if ch >= 128, we need to fall back to calling CharInClass. For ASCII, we
+            // can use a lookup table, but if it's a known set of ASCII characters we can also use a helper.
+            string asciiExpr = bitVectorString switch
+            {
+                "\0\0\0\u03ff\ufffe\u07ff\ufffe\u07ff" => $"{(negate ? "!" : "")}char.IsAsciiLetterOrDigit(ch)",
```

Comment:
These leak the details of the regex char class representation into the generator. Why not follow the pattern of RegexCharClass.LetterClass, etc and hide the strings in there? You could put them in their own file as a small partial class like RegexCharClass.InternalBitsSharedWithGeneratorAndCompiedIntoIt.cs

Reply:
It would also be more readable here.

Reply:
> These leak the details of the regex char class representation into the generator.

No, they don't. This is a bitmap created a few lines earlier.

Reply:
Oh yes! I thought these were the char class things.